app/controllers/application_controller.rb

```diff
@@ -5,7 +5,8 @@ class ApplicationController < ActionController::Base
 
   helper_method :current_user
 
-  if Rails.env.production?
+  if !Rails.application.secrets.http_auth_username.nil? &&
```

Comment:
What do you think about making this `blank?` instead of `nil?`? That way, if the key is present, but is set to an empty string, this will still work as expected.

Reply:
Or rather, instead of saying "not nil", it's better to check for presence:

```ruby
if Rails.application.secrets.http_auth_username.present? &&
  Rails.application.secrets.http_auth_password.present?
```